holoviews/plotting/bokeh/callbacks.py

```diff
@@ -92,19 +104,41 @@ class Callback(object):
     attributes = {}
 
     js_callback = """
+        function unique_events(events) {{
+            var unique = [];
+            var unique_events = [];
+            for (var i=0; i<events.length; i++) {{
+                [event, data] = comm_state.events[i];
+                if (!unique_events.includes(event)) {{
+                    unique.unshift(data);
+                    unique_events.push(event);
+                }}
+            }}
+            return unique;
+        }}
+
+        function process_events(comm_state) {{
+            var events = unique_events(comm_state.events);
```

Comment:
So this is at least one place where ``unique_events`` is used and ``comm_state`` is available. You could definitely pass ``comm_state`` to ``unique_events`` here though ``unique_events(comm_state.events, comm_state)`` is a bit redundant (just pass ``comm_state``). The fact that you can access ``comm_state`` at all in ``unique_event`` without explicitly declaring it makes me uncomfortable (but then again, this is Javascript!)